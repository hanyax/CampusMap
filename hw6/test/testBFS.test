# Simple test of breadth-first search algorithm on a small Graph build with
# tested Marvel Parser on *.tsv
# Note that in this graph, there is an edge from GORILLA-MAN to HAWK but not
# vice versa. In your full parser-generated Marvel graph, relationships are
# bidirectional, meaning an edge from A to B will always be accompanied by an
# edge from B to A.

LoadGraph heroes *.tsv

# Test longest path possible
FindPath heroes Iron_Man Thor

# Test direct connection
FindPath heroes Hulk Black_Widow

# Test no path
FindPath heroes Iron_Man Bat_Man
