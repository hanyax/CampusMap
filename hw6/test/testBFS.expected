# Simple test of breadth-first search algorithm on a small Graph build with
# tested Marvel Parser on *.tsv
# Note that in this graph, there is an edge from GORILLA-MAN to HAWK but not
# vice versa. In your full parser-generated Marvel graph, relationships are
# bidirectional, meaning an edge from A to B will always be accompanied by an
# edge from B to A.

loaded graph heroes

# Test longest path possible
path from Iron Man to Thor:
Iron Man to Black Widow via IM2
Black Widow to Thor via A4

# Test direct connection
path from Hulk to Black Widow:
Hulk to Black Widow via CA3

# Test no path
path from Iron Man to Bat Man:
no path found
